run_md.py

```diff
@@ -0,0 +1,879 @@
+
+
+def add_ligands_to_topol(all_itp_list, all_posres_list, all_resids, topol):
+    itp_include_list, posres_include_list, resid_include_list = [], [], []
+    for itp, posres, resid in zip(all_itp_list, all_posres_list, all_resids):
+        itp_include_list.append(f'; Include {resid} topology\n'
+                                f'#include "{itp}"\n')
+        posres_include_list.append(f'; {resid} position restraints\n#ifdef POSRES_{resid}\n'
+                                  f'#include "{posres}"\n#endif\n')
+        resid_include_list.append(f'{resid}             1')
+
+    edit_topology_file(topol, pattern="; Include forcefield parameters",
+                add='\n'.join(itp_include_list), how='after', n=3)
+    #reverse order since add before pattern
+    edit_topology_file(topol, pattern="; Include topology for ions",
+                add='\n'.join(posres_include_list[::-1]), how='before')
+    # ligand molecule should be after protein (or all protein chains listed)
+    n_line_after_molecules_section = get_n_line_of_protein_in_mol_section(topol_file=topol)
+    edit_topology_file(topol, pattern='[ molecules ]', add='\n'.join(resid_include_list),
+                       how='after', n=n_line_after_molecules_section)
+
+
+def prep_ligand(mol, script_path, project_dir, wdir_ligand, addH=True):
+    molid = mol.GetProp('_Name')
+    resid = mol.GetProp('resid')
+
+    wdir_ligand_cur = os.path.join(wdir_ligand, molid)
+    os.makedirs(wdir_ligand_cur, exist_ok=True)
+
+    mol_file = os.path.join(wdir_ligand_cur, f'{molid}.mol')
+
+    if os.path.isfile(os.path.join(wdir_ligand_cur, f'{molid}.itp')) and os.path.isfile(os.path.join(wdir_ligand_cur, f'posre_{molid}.itp'))\
+            and os.path.isfile(os.path.join(wdir_ligand_cur, 'resid.txt')):
+        logging.warning(f'{molid}.itp and posre_{molid}.itp files already exist. Mol preparation step will be skipped for such molecule\n')
+        with open(os.path.join(wdir_ligand_cur, 'resid.txt')) as inp:
+            molid, resid = inp.read().strip().split('\t')
+
+        return wdir_ligand_cur, molid, resid
+
+    if addH:
+        mol = Chem.AddHs(mol, addCoords=True)
+
+    Chem.MolToMolFile(mol, mol_file)
+
+    try:
+        subprocess.check_output(f'script_path={script_path} lfile={mol_file} input_dirname={wdir_ligand_cur} name={resid} bash {os.path.join(project_dir, "lig_prep.sh")}',
+                                shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(f'{molid}\t{e}\n')
+        return None
+
+    # create log for molid resid corresponding
+    with open(os.path.join(wdir_ligand_cur, 'resid.txt'), 'w') as out:
+        out.write(f'{molid}\t{resid}\n')
+    with open(os.path.join(wdir_ligand, 'all_resid.txt'), 'a') as out:
+        out.write(f'{molid}\t{resid}\n')
+
+    return wdir_ligand_cur, molid, resid
+
+def prep_md_files(var_lig_data, protein_name, system_lig_data, wdir_protein, wdir_md, clean_previous=False):
```

Comment:
add docstring where mention the strict necessity of the same order in all lists of the output dict